### PR TITLE
Fix #2311: Changes Feedback Item

### LINKED ISSUE #2311: Changes Feedback Item [A11y]
**Describe the bug**

Feedback Item is not fully compatible with some users from the Accessibility point of view.

Issue - This view has a similar issue as this one - [#2309](https://github.com/oppia/oppia-android/issues/2309)

**To Reproduce**
Steps to reproduce the behaviour:
1. Start the Accessibility Scanner App on your device
2. Open the Oppia Android App and start First Test Topic Exploration
3. Scan the screenshots with the Accessibility Scanner
4. You will see the tests that are not passed by that screen
5. Repeat this for other screens

**Expected behaviour**
All the tests must be passed by scanning the screenshots for both Tablet and Mobile Devices in every orientation.

**Screenshots**

|  |  | 
|:-------:|:-------:|
| Screenshot_2020-12-17-11-06-51-43_80a9469f53bfe75dbc6e8989be28e2f0 | Screenshot_2020-12-17-11-09-16-38_80a9469f53bfe75dbc6e8989be28e2f0 |

**Device**
- Real Device - RealMe3
- SDK version - 29

**Additional context**
The Screenshots are descriptive enough for understanding the issues but still, if there is a need of clarification then you can comment here. Multiple Pull Requests can also be made for solving these issues.

Certain Views does not have the correct contrast ratios (Inactive state) as mentioned in the screenshots. To solve this, one must find a way to make the views disabled instead of being in non-clickable state only

Reference - [https://github.com/oppia/oppia-android/wiki/Working-on-UI#accessibility-scanner](https://github.com/oppia/oppia-android/wiki/Working-on-UI#accessibility-scanner)

### PR DESCRIPTION
**Pull request by Mehul-Bisht**
## Explanation

Fix #2311: Changes Feedback Item
Fixes the accessibility issues in feedback item in all layouts.

## Portrait Mode
| Before | After |
:-------------------------:|:-------------------------:
portrait_before_feedback | portrait_after_feedback

## Landscape Mode
| Status |
:-------------------------:
## Before changes
land_before_feedback
## After changes
land_after_feedback

## Checklist

- [x] The PR title starts with "Fix #bugnum: ", followed by a short, clear summary of the changes. (If this PR fixes part of an issue, prefix the title with "Fix part of #bugnum: ...".)
- [x] The PR explanation includes the words "Fixes #bugnum: ..." (or "Fixes part of #bugnum" if the PR only partially fixes an issue).
- [x] The PR follows the [style guide](https://github.com/oppia/oppia-android/wiki/Coding-style-guide).
- [x] The PR does not contain any unnecessary auto-generated code from Android Studio.
- [x] The PR is made from a branch that's **not** called "develop".
- [x] The PR is made from a branch that is up-to-date with "develop".
- [x] The PR's branch is based on "develop" and not on any other branch.
- [x] The PR is **assigned** to an appropriate reviewer in both the **Assignees** and the **Reviewers** sections.

### CODE REVIEW

**Comment by Arjupta** [comment]
@Mehul-Bisht can you describe the need for the changes in `content_item` files , does solving this Issue required that changes too 
OR
You meant to solve two Issues at a time. If this is the case then can you make separate PR for all the changes related to `content_item_view` , as this #2309 tracks this particular view

**Comment by Mehul-Bisht** [comment]
@ARJUPTA  oh I didn't notice a separate issue existed for the content view, I'll do the necessary changes over there soon!

**Comment by Mehul-Bisht** [comment]
@ARJUPTA kindly check again now

File: app/src/main/res/values/dimens.xml [hunk 1]
```diff
@@ -165,8 +165,10 @@
   <dimen name="content_item_exploration_view_padding_top">12dp</dimen>
   <dimen name="content_item_question_split_view_margin_bottom">12dp</dimen>
   <dimen name="content_item_question_view_margin_bottom">12dp</dimen>
-  <dimen name="feedback_item_exploration_view_padding_bottom">12dp</dimen>
-  <dimen name="feedback_item_exploration_view_padding_top">12dp</dimen>
+  <dimen name="feedback_item_exploration_view_padding_bottom">0dp</dimen>
+  <dimen name="feedback_item_exploration_view_padding_top">0dp</dimen>
+  <dimen name="feedback_text_view_padding_top">13dp</dimen>
+  <dimen name="feedback_text_view_padding_bottom">13dp</dimen>
```
Comment (Arjupta) [inline app/src/main/res/values/dimens.xml]: @Mehul-Bisht shouldn't we have top and bottom padding to be 12dp here?
`  <dimen name="feedback_text_view_padding_top">13dp</dimen>`
 ` <dimen name="feedback_text_view_padding_bottom">13dp</dimen>`

File: app/src/main/res/values/dimens.xml [hunk 2]
```diff
@@ -247,8 +249,10 @@
   <dimen name="drag_drop_grouping_hint_non_split_text_view_margin_top">8dp</dimen>
   <dimen name="drag_drop_container_parent_non_split_margin_end">8dp</dimen>
   <dimen name="drag_drop_container_parent_non_split_margin_start">8dp</dimen>
-  <dimen name="feedback_item_exploration_view_padding_end">8dp</dimen>
-  <dimen name="feedback_item_exploration_view_padding_start">8dp</dimen>
+  <dimen name="feedback_item_exploration_view_padding_end">0dp</dimen>
+  <dimen name="feedback_item_exploration_view_padding_start">0dp</dimen>
+  <dimen name="feedback_text_view_padding_start">8dp</dimen>
+  <dimen name="feedback_text_view_padding_end">8dp</dimen>
```
Comment (Arjupta) [inline app/src/main/res/values/dimens.xml]: The names of the text view padding should be started with `feedback_item_text_view` to give more context about its use.

**Review by Arjupta — CHANGES_REQUESTED** [review]

**Comment by Mehul-Bisht** [comment]
@ARJUPTA i have made the changes, please check

**Review by rt4914 — CHANGES_REQUESTED** [review]
Please have a look at my suggestion mentioned in #2558 
Those same suggestions apply here too.

**Comment by Arjupta** [comment]
> @ARJUPTA i have made the changes, please check

@Mehul-Bisht try this solution for both PRs https://github.com/oppia/oppia-android/pull/2558#issuecomment-767466874

File: app/src/main/res/values/dimens.xml (hunk 1 above)
Comment (Mehul-Bisht) [reply app/src/main/res/values/dimens.xml]: @ARJUPTA I had to make this to 13 dp to fix the issue
padding

**Comment by rt4914** [comment]
Replied on https://github.com/oppia/oppia-android/pull/2558

**Comment by Luffy18346** [comment]
@Mehul-Bisht any updates on this?

**Comment by Mehul-Bisht** [comment]
> @Mehul-Bisht any updates on this?

@Luffy18346 Was working on other issues since then, will take a look today again and let you know if I'm able to fix it completely.

**Comment by Luffy18346** [comment]
> > @Mehul-Bisht any updates on this?
> 
> @Luffy18346 Was working on other issues since then, will take a look today again and let you know if I'm able to fix it completely.

Ok, thanks.

**Comment by Mehul-Bisht** [comment]
@rt4914 @ARJUPTA I have fixed the issue, kindly review

**Comment by Mehul-Bisht** [comment]
@rt4914 @ARJUPTA any further updates from you on the latest changes?

**Comment by Arjupta** [comment]
> @rt4914 @ARJUPTA any further updates from you on the latest changes?

Testing it

**Review by Arjupta — CHANGES_REQUESTED** [review]
|This PR | Develop |
|:-:|:-:|
|Screenshot_1612351526|Screenshot_1612351689|

The displacement issue is is still not solved completely

**Comment by rt4914** [comment]
Should be done once my set of PRs gets merged. Reference: https://github.com/oppia/oppia-android/pull/2558#issuecomment-772576421

**Comment by BenHenning** [comment]
@Mehul-Bisht please update this with the latest changes from develop. I think you have some conflicts to resolve.

**Comment by rt4914** [comment]
Closing this, as this will get covered once the entire Highfi work of Explorations/Question player is finished.

**Pull request closed without merging**